source/menu.cpp

```diff
@@ -62,57 +81,45 @@ void titles_multkey_draw(const char *title, const char* footer, int back, std::v
-    int previous_menu_offset = 1;
     int menu_pos_y;
-    int menu_end_y = currentMenu.menuConsole.consoleHeight -1; 
+    int menu_end_y = 19;
     int current_pos_y = 0;
 
-    while (aptMainLoop()) {
+    while (!bExit && aptMainLoop()) {
         if(firstLoop || previous_index != current) {
             firstLoop = false;
             int results_per_page = menu_end_y - menu_pos_y;
             int current_page = current / results_per_page;
             menu_offset = current_page * results_per_page;
-            char prev_entry[50];
-            char entry[50];
-            sprintf(prev_entry, "%-*.*s (%s) %*.*s",
-                31, 31, (*options)[previous_index].name.c_str(),
-                (*options)[previous_index].region.c_str(),
-                10, 10, (*options)[previous_index].code.c_str());
-            sprintf(entry, "%-*.*s (%s) %*.*s",
-                31, 31, (*options)[current].name.c_str(),
-                (*options)[current].region.c_str(),
-                10, 10, (*options)[current].code.c_str());
-            if(menu_offset == previous_menu_offset) {
-                menu_draw_string(prev_entry, 1, menu_pos_y + (previous_index-menu_offset), CONSOLE_WHITE);
-                menu_draw_string(entry, 1, menu_pos_y + (current-menu_offset), CONSOLE_REVERSE);
-            } else {
-                consoleClear();
-                current_pos_y=0;
-                // Draw the header
-                menu_draw_string(title, 0, current_pos_y++, CONSOLE_RED);
-                menu_pos_y = current_pos_y;
-                for (int i = 0; menu_offset + i < count && i < results_per_page; i++) {
-                    sprintf(entry, "%-*.*s (%s) %*.*s",
-                        31, 31, (*options)[menu_offset + i].name.c_str(),
-                        (*options)[menu_offset + i].region.c_str(),
-                        10, 10, (*options)[menu_offset + i].code.c_str());
-                    if(i+menu_offset == current) {
-                        menu_draw_string(entry, 1, current_pos_y, CONSOLE_REVERSE);
-                    } else {
-                        menu_draw_string(entry, 1, current_pos_y, CONSOLE_WHITE);
-                    }
-                    current_pos_y++;
-                    previous_menu_offset = menu_offset;
-                }
-                if (footer != NULL)
-                {
-                    // Draw the footer if one is provided
-                    current_pos_y = currentMenu.menuConsole.consoleHeight - 1;
-                    menu_draw_string_full(footer, current_pos_y, CONSOLE_BLUE CONSOLE_REVERSE);
+            current_pos_y=0;
+            // Draw the header
+            setTextColor(COLOR_TITLE);
+            renderText(0, 0, 0.7f, 0.7f, false, title);
+            menu_draw_string(title, 0, current_pos_y++, CONSOLE_RED);
+            menu_pos_y = current_pos_y;
+            for (int i = 0; menu_offset + i < count && i < results_per_page; i++) {
+                int y_pos = 18+(i*12);
```

Comment:
Some more magic numbers in this function, along with menu_end_y above which used to be based on console height. Hopefully we can make these dynamically determined :)